### DISCUSSION TITLE
GPU compatible neper

### DISCUSSION THREAD

**Discussion by rsooraj** [neperfepx/neper#899] [Q&A] [answered]
I am an end user of Neper. 
The time taken for polycrystal generation was huge for reasonably large sample sizes. 
Is there a GPU version (CUDA or openacc) of Neper available? 
Is it possible to enable GPU offloading for Neper?
I tried little with the source code, to change openmp pragma statements to openacc. 
This ended up in some error. 
Then I tried tracing the functions for profiling, without success. 
Is there any guidelines or suggestions?

**Comment by rquey** [comment, marked as answer]
Hmm...  How huge is it, and what is the sample size? and what is your Neper command? You may be running in sub-optimal conditions...

There is no GPU implementation of Neper.

**Comment by rsooraj** [comment]
Thank you for the response.
I was trying to generate a polycrystal with a grain size distribution of lognormal(mu=0.1mm, sigma=0.04) on a sample of 10mmX15mmX15mm. 
The Neper command I used was
```
neper -T -n from_morpho -domain "cube(10,15,15)" -morpho "diameq:lognormal(mean=0.1,sigma=0.04),1-sphericity:lognormal(0.145,0.03)" -morphooptistop itermax=1000000 -statcell diameq,body,npolynb,sphericity,convexity,nseednb -o c_10x15x15_log_mu_1e-1_10Liter -oridescriptor euler-bunge -format ply,tess
```

What might be the possible difficulties of porting it to GPU?

**Reply by rquey** [reply]
This is almost 3 millions cells. Neper's algorithms take a very long time for such numbers.

I've just never done GPU myself.

**Comment by rsooraj** [comment]
What about using openacc instead of openmp pragmas?

**Comment by rsooraj** [comment]
Thank you for the comments.
